lib/puppet/provider/archive/ruby.rb

```diff
@@ -1,13 +1,5 @@
-begin
-  require 'puppet_x/bodeco/archive'
-  require 'puppet_x/bodeco/util'
-rescue LoadError
-  require 'pathname' # WORK_AROUND #14073 and #7788
-  archive = Puppet::Module.find('archive', Puppet[:environment].to_s)
-  raise(LoadError, "Unable to find archive module in modulepath #{Puppet[:basemodulepath] || Puppet[:modulepath]}") unless archive
-  require File.join archive.path, 'lib/puppet_x/bodeco/archive'
-  require File.join archive.path, 'lib/puppet_x/bodeco/util'
-end
+require File.dirname(__FILE__) + '/../../../puppet_x/bodeco/archive.rb'
```

Comment:
Since we are ruby 1.9.3+, we can just use require_relative. The patch here might appear to 'fix' the problem, but it's likely coincidental. The core problem is https://tickets.puppetlabs.com/browse/SERVER-94.